tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -83,26 +83,30 @@ export const Manufacturer = (_, context) => {
   }
 
   // Get a ManufacturableData from a QueueBlueprint using its type, category, and name.
-  let queueBlueprintRefs = data.queue.map((queued:QueueBlueprint) =>
+  let queueBlueprintRefs = data.queue && data.queue.length
+  && data.queue.map((queued:QueueBlueprint) =>
     blueprints_by_category[queued.category].find((key) => (key.name === queued.name))
   );
 
   return (
     <Window width={1200} height={600} title={data.fabricator_name}>
-      {!data.indicators.hasPower && <PowerAlertModal width={100-SETTINGS_WINDOW_WIDTH} height={"100%"} />}
+      {data.indicators && !data.indicators.hasPower
+      && <PowerAlertModal width={100-SETTINGS_WINDOW_WIDTH} height={"100%"} />}
```

Comment:
Huh, surprised I didn't catch this earlier. This is a weird way of doing width. Will revisit at a different time.

Reply:
Uhh sounds good, this just represents the percentage of the window that is not the settings bar though. 100%-20%.